### PR TITLE
mon/MgrStatMonitor: do not crash on luminous dev version upgrades

### PR DESCRIPTION
**Pull request by liewegas**
Signed-off-by: Sage Weil <sage@redhat.com>

### CODE REVIEW

**Comment by ukernel** [comment]
LGTM

**Comment by jecluis** [comment]
fwiw, I hit this issue and upon cherry-picking this patch everything went back to working seamlessly.

File: src/mon/MgrStatMonitor.cc [hunk 1]
```diff
@@ -85,13 +85,18 @@ void MgrStatMonitor::update_from_paxos(bool *need_bootstrap)
...
+      auto p = bl.begin();
+      ::decode(digest, p);
+      ::decode(health_summary, p);
+      ::decode(health_detail, p);
+      ::decode(service_map, p);
+      dout(10) << __func__ << " v" << version
+	       << " service_map e" << service_map.epoch << dendl;
+    }
+    catch (buffer::error& e) {
+      derr << "failed to decode mgrstat state; luminous dev version?" << dendl;
```
Comment (gregsfortytwo) [inline src/mon/MgrStatMonitor.cc]: Can we do some kind of tagging so we know that's what happened going forward?

Or are all the upgrade paths certain to be safe with default values of those members? Given the previous discussion on its uses I'm a little worried about losing the service map.

**Comment by liewegas** [comment]
This is only intended to ease the upgrade path. I can try to update it to ignore during update but assert out otherwise?

During upgrade we don't care about losing anything here since it's all new anyway.

**Comment by gregsfortytwo** [comment]
Yeah I'm just worried that the way it's formulated we could (basically silently) lose data long after the upgrades are done.

**Pull request merged**